src/components/createElementComponent.tsx

```diff
@@ -69,15 +67,21 @@ const createElementComponent = (
     onShippingAddressChange,
     onShippingRateChange,
   }) => {
-    const {elements} = useElementsContextWithUseCase(`mounts <${displayName}>`);
+    const ctx = useElementsOrCustomCheckoutSdkContextWithUseCase(
+      `mounts <${displayName}>`
+    );
+    const elements = 'elements' in ctx ? ctx.elements : null;
+    const customCheckoutSdk =
+      'customCheckoutSdk' in ctx ? ctx.customCheckoutSdk : null;
     const [element, setElement] = React.useState<stripeJs.StripeElement | null>(
       null
     );
     const elementRef = React.useRef<stripeJs.StripeElement | null>(null);
     const domNode = React.useRef<HTMLDivElement | null>(null);
 
     const {setCart, setCartState} = useCartElementContextWithUseCase(
-      `mounts <${displayName}>`
+      `mounts <${displayName}>`,
+      'customCheckoutSdk' in ctx
```

Comment:
`<CustomCheckoutProvider />` does NOT have any Cart related stuff.
In order to avoid changing the interface of `createElementComponent` so that users can still use `<PaymentElement />`, this is to check whether it's in the context of CustomCheckoutProvider or Elements.
Note `customCheckoutSdk` can be null because it's **async** set in the provider.